### PR TITLE
Update remotes

### PR DESCRIPTION
**Pull request by jimhester**
Implementation of `update_remotes` as defined in #1037, #975.

I also improved the SVN support to include the SVN revision, so this will now work with all remotes in devtools.

I still need to add tests for this, but this works with a simple test on all remote types on my machine.

### CODE REVIEW

File: R/update-remote.R [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    if (!yesno("Are you sure you want to update all installed packages with remotes?")) {
+      pkgs <- installed.packages()[, "Package"]
+    } else {
+      return(invisible())
+    }
+  }
+
+  remotes <- compact(lapply(pkgs, package2remote))
+
+  if (!isTRUE(force)) {
```
Comment (hadley) [inline R/update-remote.R]: Is there are reason to do this here, rather than relying on the individual install methods?

**Comment by jimhester** [comment]
https://github.com/jimhester/devtools/commit/ed000dd17f5561464915b28d2d5d3c2a3e0c3caa does as you suggest moving the SHA checking into `install_remote()` so it works the same across all remote types.

I also moved the remote updating into `update.package_deps`, so it will automatically work with the existing `update_packages()`.

File: R/install-remote.R [hunk 2]
```diff
@@ -108,3 +116,46 @@ remote_download <- function(x, quiet = FALSE) UseMethod("remote_download")
 remote_metadata <- function(x, bundle = NULL, source = NULL) UseMethod("remote_metadata")
 remote_package_name <- function(remote, ...) UseMethod("remote_package_name")
 remote_sha <- function(remote, ...) UseMethod("remote_sha")
+
+package2remote <- function(x, ...) {
+  x <- packageDescription(x)
+  if (is.null(x$RemoteType)) {
+    return(NULL)
```
Comment (hadley) [inline R/install-remote.R]: What do you think about having a CRAN package type here? That might make it easy for `update_packages()` to work with everything

Comment (jimhester) [inline R/install-remote.R]: This has been done at https://github.com/jimhester/devtools/commit/8bdb10bb92972cc26d02a0d1ef6c711c24bbec, let me know your thoughts...

File: R/install-CRAN.r [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+#' @param pkgs Character vector of packages to install.
+#' @inheritParams package_deps
+#' @export
+#' @family package installation
+#' @examples
+#' \dontrun{
+#' install_CRAN("ggplot2")
+#' install_CRAN(c("httpuv", "shiny")
+#' }
+install_CRAN <- function(pkgs, repos = getOption("repos"), type = getOption("pkgType"), ..., quiet = FALSE) {
```
Comment (hadley) [inline R/install-CRAN.r]: I'd prefer all lowercase to be consistent

File: R/install-CRAN.r [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+    name = pkg,
+    repos = repos,
+    pkg_type = type)
+}
+
+
+#' @export
+remote_download.CRAN_remote <- function(x, quiet = FALSE) {
+  dest_dir <- tempdir()
+  download.packages(x$name, destdir = dest_dir, repos = x$repos, type = x$pkg_type)[1, 2]
```
Comment (hadley) [inline R/install-CRAN.r]: I forget how this works, but are we already computing the dependencies that are missing/out-of-date?

Comment (jimhester) [inline R/install-CRAN.r]: Dependencies are all done in `install()` itself which calls `install_deps()` so we don't have to worry about it here.

File: R/install-CRAN.r (hunk 3 above)
Comment (jimhester) [inline R/install-CRAN.r]: Done at https://github.com/jimhester/devtools/commit/16dbe00e4bc70bcd3eb0d2e09e2c221c39860715

**Comment by krlmlr** [comment]
While you're at it: Is there a way to *not* install/upgrade remotes, just to install a single package? I tried

```
devtools::install("path", dependencies = FALSE, upgrade_dependencies = FALSE, metadata = NULL)
```

to no avail. (Now I know that the "metadata" arg does something different, never mind.) To me, it looks like install_dev_remotes() is called unconditionally. For now, I'll be using `RCMD("INSTALL")` as workaround.

**Comment by jimhester** [comment]
@krlmlr You are correct, the remotes should respect `update_dependencies` as well.

**Comment by jimhester** [comment]
https://github.com/hadley/devtools/pull/1067/commits/c35630ca941b043b8560c63801ab7a34ac13507d has a simple implementation to handle #1109

**Pull request merged**